mkosi

```diff
@@ -492,14 +536,15 @@ def determine_partition_table(args: CommandLineArguments):
     return table, run_sfdisk
 
 
-def create_image(args: CommandLineArguments, workspace: str, for_cache: bool) -> Optional[IO[str]]:
+def create_image(args: CommandLineArguments, workspace: str, for_cache: bool) -> Optional[BinaryIO]:
     if not args.output_format.is_disk():
         return None
 
     with complete_step('Creating partition table',
                        'Created partition table as {.name}') as output:
 
-        f = tempfile.NamedTemporaryFile(dir=os.path.dirname(args.output), prefix='.mkosi-', delete=not for_cache)
+        f: BinaryIO = cast(BinaryIO, tempfile.NamedTemporaryFile(prefix='.mkosi-', delete=not for_cache,
```

Comment:
Doesn't the cast tell the inference engine enough about `f` that you can drop the type annotation entirely?

Reply:
Yes.

My general mindset is "every use of `cast()` is a workaround for a bug/deficiency in mypy, and should be able to be dropped as mypy improves."  So, eventually, when mypy improves, I think the line should be:

    f: BinaryIO = tempfile.NamedTemporaryFile(prefix='.mkosi-', delete=not for_cache,

So, I included the `: BinaryIO` in it, since I think it should be there when the `cast()` is (hopefully) eventually dropped; even though it isn't necessary as long as the `cast()` is present.

Reply:
Yes.

My mindset is "every call to `cast()` is a workaround for a bug/deficiency in mypy, and should be droppable as mypy improves."

I think the "correct" way to write the line is

    f: BinaryIO = tempfile.NamedTemporaryFile(prefix='.mkosi-', delete=not for_cache,

Since I'd like the `: BinaryIO` to be there after the `cast()` is removed, I included it now, even though it isn't necessary as long as `cast()` is there.